### DISCUSSION TITLE
change filament

### DISCUSSION THREAD

**Discussion by Mario-vminion** [OrcaSlicer/OrcaSlicer#179] [General]
i want to change the filament during the print... i see there is pause function on a layer...when i select this will the print head park to a save place that i can change the filament ...

**Comment by Pmant** [comment]
Usually the slicer would tell the printer to pause with G-code M601 or M25 and it's up to the the printer's firmware to do the rest. 
Go to the printer settings -> **enable advanced** -> **Machine G-code** -> **Pause G-code**. 
This is the G-code the slicer will use for pauses. Most printer firmwares support M601 or M25 to pause the print and move the printhead somewhere. Otherwise you could manually add G-code to the **Pause G-code** field.

**Reply by Baloo3D** [reply]
Thank you for your post! I understand that the slicer typically uses commands like M601 or M25 to pause the print, and that it's the printer's firmware that handles the subsequent actions. I've already checked the printer settings and enabled advanced options, but the issue persists.
In OrcaSlicer, the Pause G-code field does not seem to generate the necessary commands automatically to instruct the extruder on its position before pausing. While I can manually add G-code to this field, I've found that using G92 E0 does not work with my printer, as it causes the extruder to stop injecting filament after the pause.
I’ve had no issues with M25 in Cura, where it generates the required code to maintain the extruder's position, but it seems OrcaSlicer lacks this feature. I really appreciate any further insights or suggestions you might have to help resolve this! Thank you again!
